src/service/resource.js

```diff
@@ -357,10 +365,18 @@ export class Resource {
     const viewport = this.resources_.getViewport();
     this.layoutBox_ = box;
 
+    const win = this.resources_.win;
+    const computedStyle = win./*OK*/getComputedStyle(this.element);
```

Comment:
Yes, it has to be in the `vsync.measure`. I think we need to change this. Basically, we only need to read margins in a relatively rare case when margin change has been requested: they are not needed before or after otherwise. So, what we can do, is remove `layoutMargins_` var and code from `Resource` file and modify this code here: https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1334

It'd be something like this:
```
if (resource.hasBeenMeasured() && !opt_newMargings) {
  // unchanged
} else {
  vsync.measure(() => {
    if (!resource.hasBeenMeasured()) {
     resource.measure();
    }
    let marginsDiff = null;
    if (opt_newMargin) {
      marginsDiff = // calculate existing margins and diff with opt_newMargin
    }
    this.completeScheduleChangeSize_(..., marginsDiff);
  });
}
```

And then we'd use `marginsDiff` in the resize code itself. E.g. `marginsDiff && marginsDiff.top` would mean that top margin has been modified, etc.